src/Routing/Hash.purs

```diff
@@ -6,14 +6,15 @@ import DOM
 foreign import setHash """
 function setHash(hash) {
   return function() {
-    document.location.hash = hash;
+    var uri = document.location.href.split('#')[0];
```

Comment:
Perhaps use `location.href.substring(location.href.indexOf("#") + 1)` kind of thing here instead, in case there is a second hash in the hash?